### PR TITLE
Rename ignoreNil() to skipNil()

### LINKED ISSUE #2990: Rename ignoreNil to skipNil?
I was thinking about some RAC operators yesterday and realized that we use both `ignore` and `skip` to skip events. These do the same-ish thing, so they should probably use the same name. `ignoreNil` does change the type of `Value`, but I'd prefer to think of that as type-refinement.

Likewise, if we add `ignoreErrors`, it should probably be called `skipErrors`.

### PR DESCRIPTION
**Pull request by mdiep**
Resolves #2990.

May require some consensus.

### CODE REVIEW

**Comment by NachoSoto** [comment]
Yup, agreed :+1:

**Comment by neilpa** [comment]
I'll throw out one other alternative of `dropNil()`

**Comment by mdiep** [comment]
I'm not a fan of the `drop` prefix, even though it would match the Swift stdlib. (Note that all the `skip*` methods would need to be renamed.)

This is ready to go now as I've fixed the failing tests.

File: ReactiveCocoaTests/Swift/SignalSpec.swift [hunk 1]
```diff
@@ -400,7 +400,7 @@ class SignalSpec: QuickSpec {
 		describe("ignoreNil") {
```
Comment (NachoSoto) [inline ReactiveCocoaTests/Swift/SignalSpec.swift]: Needs to change this too :)

File: ReactiveCocoaTests/Swift/SignalProducerLiftingSpec.swift [hunk 2]
```diff
@@ -77,7 +77,7 @@ class SignalProducerLiftingSpec: QuickSpec {
 		describe("ignoreNil") {
```
Comment (NachoSoto) [inline ReactiveCocoaTests/Swift/SignalProducerLiftingSpec.swift]: And this

**Comment by NachoSoto** [comment]
:shipit: after those last 2 things. I think this name is much better, consistent with the other skip operators.

**Comment by mdiep** [comment]
🚢

**Pull request merged**